### PR TITLE
Accuracy Checker: fix cpu_extension search for Mac OS

### PR DESCRIPTION
**Pull request by eaidova**
@IRDonch, please take a look. 
P.S. Internal validation pipeline passed.

### CODE REVIEW

File: tools/accuracy_checker/accuracy_checker/launcher/dlsdk_launcher.py [hunk 1]
```diff
@@ -330,7 +330,19 @@ def get_cpu_extension(cpu_extensions, selection_mode):
         if cpu_extensions_name != 'AUTO':
             return cpu_extensions
         extensions_path = cpu_extensions.parent
-        file_format = '{}.dll' if platform.system() == 'Windows' else 'lib{}.so'
+        os_specific_formats = {
+            'Windows': '{}.dll',
+            'Darwin': 'lib{}d.dylib',
```
Comment (IRDonch) [inline tools/accuracy_checker/accuracy_checker/launcher/dlsdk_launcher.py]: This should be `lib{}.dylib`. The `d` version is the debug build, I believe.

Also, I would suggest to list these in alphabetical order and to add a trailing comma.

File: tools/accuracy_checker/accuracy_checker/launcher/dlsdk_launcher.py [hunk 2]
```diff
@@ -330,7 +330,19 @@ def get_cpu_extension(cpu_extensions, selection_mode):
...
+            'Linux': 'lib{}.so'
+        }
+        system_name = platform.system()
+        file_format = os_specific_formats.get(system_name)
+        if not file_format:
+            raise ConfigError(
+                'Accuracy Checker can not automatically found cpu extensions library '
```
Comment (IRDonch) [inline tools/accuracy_checker/accuracy_checker/launcher/dlsdk_launcher.py]: find*

Comment (eaidova) [reply tools/accuracy_checker/accuracy_checker/launcher/dlsdk_launcher.py]: fixed

File: tools/accuracy_checker/accuracy_checker/launcher/dlsdk_launcher.py (hunk 1 above)
Comment (eaidova) [reply tools/accuracy_checker/accuracy_checker/launcher/dlsdk_launcher.py]: > The d version is the debug build, I believe.
Most likely. I also add additional search loop if a release lib is not found.

Comment (IRDonch) [reply tools/accuracy_checker/accuracy_checker/launcher/dlsdk_launcher.py]: > Most likely. I also add additional search loop if a release lib is not found.

Why would it not be found?

Comment (eaidova) [reply tools/accuracy_checker/accuracy_checker/launcher/dlsdk_launcher.py]: Possibly, if a user builds IE from sources in debug mode, he can not have release lib (but I do not sure about it...)

Comment (IRDonch) [reply tools/accuracy_checker/accuracy_checker/launcher/dlsdk_launcher.py]: Fair enough. However, macOS support should go into R2, while I would prefer to postpone the debug mode support until R3, to minimize the potential for regressions. So let's stick with just enabling macOS in this PR, OK?

Comment (IRDonch) [reply tools/accuracy_checker/accuracy_checker/launcher/dlsdk_launcher.py]: (FWIW, I don't think searching for either version of the library makes sense anyway. The extension library should match the main IE library.)

Comment (eaidova) [reply tools/accuracy_checker/accuracy_checker/launcher/dlsdk_launcher.py]: Ok, I remove debug.

File: tools/accuracy_checker/accuracy_checker/launcher/dlsdk_launcher.py [hunk 3]
```diff
@@ -330,27 +330,50 @@ def get_cpu_extension(cpu_extensions, selection_mode):
...
-
-            cpu_info_flags = get_cpu_info()['flags']
-            supported_flags = ['avx512', 'avx2', 'sse4']
-            for flag in supported_flags:
-                selection_mode = flag
-                if selection_mode in cpu_info_flags:
-                    break
-        extension_list = list(extensions_path.glob(file_format.format('cpu_extension_{}'.format(selection_mode))))
+        os_specific_formats = {
+            'Darwin': 'lib{}.dylib',
```
Comment (alalek) [inline tools/accuracy_checker/accuracy_checker/launcher/dlsdk_launcher.py]: BTW, macOS loadable modules (plugins) [should have `.so` extension](https://gitlab.kitware.com/cmake/cmake/blob/v3.14.1/Modules/Platform/Darwin.cmake#L34-38) (internal bug has been filled for DLDT), so it make sense to add support for multiple variants.

Comment (eaidova) [reply tools/accuracy_checker/accuracy_checker/launcher/dlsdk_launcher.py]: What should have higher priority `.so` or `dylib`?

Comment (eaidova) [reply tools/accuracy_checker/accuracy_checker/launcher/dlsdk_launcher.py]: ok, I add multiple variants

Comment (IRDonch) [reply tools/accuracy_checker/accuracy_checker/launcher/dlsdk_launcher.py]: I don't think it matters. There shouldn't ever be both.

File: tools/accuracy_checker/accuracy_checker/launcher/dlsdk_launcher.py [hunk 4]
```diff
@@ -326,31 +326,59 @@ def _prepare_bitstream_firmware(self, config):
...
+                default_cpu_extension = file_format.format(base_name)
+                extension_list = list(extensions_path.glob(default_cpu_extension))
+
+                if extension_list:
+                    return extension_list
+
+                cpu_info_flags = get_cpu_info()['flags']
+                supported_flags = ['avx512', 'avx2', 'sse4']
+                for flag in supported_flags:
+                    if flag in cpu_info_flags:
```
Comment (IRDonch) [inline tools/accuracy_checker/accuracy_checker/launcher/dlsdk_launcher.py]: Hold on. `py-cpuinfo` doesn't actually support a flag named `sse4`. It has `sse4_1` and `sse4_2`. So this change is going to break extension library selection on CPUs that don't support AVX2.

There could probably be a better fix for this, but for now, let's just revert to the original logic here. I don't want any regressions.

File: tools/accuracy_checker/accuracy_checker/launcher/dlsdk_launcher.py [hunk 5]
```diff
@@ -326,31 +326,59 @@ def _prepare_bitstream_firmware(self, config):
...
+        if not file_formats:
+            raise ConfigError(
+                'Accuracy Checker can not automatically find cpu extensions library '
+                'for {} platform. Please, set cpu extension library manually.'.format(system_name)
+            )
 
-            if extension_list:
-                return extension_list[0]
+        if not isinstance(file_formats, (list, tuple)):
+            file_formats = [file_formats]
```
Comment (IRDonch) [inline tools/accuracy_checker/accuracy_checker/launcher/dlsdk_launcher.py]: This feels like a workaround for a self-inflicted problem. Just make every value in `os_specific_formats` a tuple.

File: tools/accuracy_checker/accuracy_checker/launcher/dlsdk_launcher.py [hunk 6]
```diff
@@ -326,31 +326,59 @@ def _prepare_bitstream_firmware(self, config):
 
     @staticmethod
     def get_cpu_extension(cpu_extensions, selection_mode):
+        def get_cpu_extensions_list(file_format, base_name, selection_mode='',):
```
Comment (IRDonch) [inline tools/accuracy_checker/accuracy_checker/launcher/dlsdk_launcher.py]: The default value is never used here, so it's a bit pointless.

File: tools/accuracy_checker/accuracy_checker/launcher/dlsdk_launcher.py (hunk 5 above)
Comment (eaidova) [reply tools/accuracy_checker/accuracy_checker/launcher/dlsdk_launcher.py]: fixed

File: tools/accuracy_checker/accuracy_checker/launcher/dlsdk_launcher.py (hunk 6 above)
Comment (eaidova) [reply tools/accuracy_checker/accuracy_checker/launcher/dlsdk_launcher.py]: fixed

File: tools/accuracy_checker/accuracy_checker/launcher/dlsdk_launcher.py (hunk 4 above)
Comment (eaidova) [reply tools/accuracy_checker/accuracy_checker/launcher/dlsdk_launcher.py]: I add a mapping from cpu_info flags to library suffixes, however, I do not know, has it ever worked on platforms without sse4_1 & sse4_2

Comment (eaidova) [reply tools/accuracy_checker/accuracy_checker/launcher/dlsdk_launcher.py]: and also return previously used logic too (for sse4 as default suffix)

**Pull request merged**